Modules/Filtering/ImageIntensity/include/itkSymmetricEigenAnalysisImageFilter.h

```diff
@@ -277,7 +280,7 @@ class SymmetricEigenAnalysisImageFilter
 #endif
 
 protected:
-  SymmetricEigenAnalysisImageFilter() = default;
+  SymmetricEigenAnalysisImageFilter() { this->SetDimension(TInputImage::ImageDimension); }
```

Comment:
Several other ITK filter classes define and set
```cpp
static constexpr unsigned int ImageDimension = TInputImage::ImageDimension;
```
If that is useful in this case -- e.g., so that the `GetDimension` method need not call to `this->GetFunctor().GetDimension()` -- then we could leverage that into this edit too, using `Self::ImageDimension` instead of `TInputImage::ImageDimension`.

Reply:
If I follow, correctly the suggestion is to move the initialization down into the functor, which would be nice. However, in this case, looking into the details, the functor is no longer templated over the image type (it just has the pixel type), losing the information on the image dimension.